charts/prometheus/templates/server/deploy.yaml

```diff
@@ -94,13 +101,12 @@ spec:
           {{- range .Values.server.extraFlags }}
             - --{{ . }}
           {{- end }}
-          {{- if .Values.server.baseURL }}
-            - --web.external-url={{ .Values.server.baseURL }}
-          {{- end }}
-
           {{- range $key, $value := .Values.server.extraArgs }}
             - --{{ $key }}={{ $value }}
           {{- end }}
+          {{- if .Values.server.baseURL }}
+            - --web.external-url={{ .Values.server.baseURL }}
+          {{- end }}
```

Comment:
These moves are not necessary.

Reply:
Compare the two files (`sts.yaml` and `deploy.yaml`) where problem sits. As stated in comment, another option is to move most parts to a named template so no more alignment/sync issues between both. Or a single templated file with several `if` :thinking:  ?